test/challtestsrv/tlsalpnone.go

```diff
@@ -47,7 +85,7 @@ func (s *ChallSrv) GetTLSALPNChallenge(host string) (string, bool) {
 func (s *ChallSrv) ServeChallengeCertFunc(k *ecdsa.PrivateKey) func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
 	return func(hello *tls.ClientHelloInfo) (*tls.Certificate, error) {
 		if len(hello.SupportedProtos) != 1 || hello.SupportedProtos[0] != va.ACMETLS1Protocol {
-			return nil, fmt.Errorf("ALPN failed, ClientHelloInfo.SupportedProtos: %s", hello.SupportedProtos)
```

Comment:
@rolandshoemaker correctly pointed out that this design removes an error that is valuable for integration tests of TLS-ALPN-01 challenges.

Consensus is that its OK to take this small hit to our confidence in the TLS-ALPN-01 integration test to land this PR, and we'll fix it in a follow-up: https://github.com/letsencrypt/boulder/issues/3962